casper/src/test/scala/coop/rchain/casper/util/rholang/DeployIdTest.scala

```diff
@@ -0,0 +1,101 @@
+package coop.rchain.casper.util.rholang
+
+import cats.effect.Resource
+import coop.rchain.casper.protocol.DeployData
+import coop.rchain.casper.util.ConstructDeploy
+import coop.rchain.casper.util.rholang.Resources._
+import coop.rchain.crypto.PrivateKey
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.Expr.ExprInstance.GBool
+import coop.rchain.models.Validator.Validator
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.models.{GDeployId, Par}
+import coop.rchain.rholang.interpreter.Runtime.BlockData
+import coop.rchain.rholang.interpreter.accounting
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.duration._
+
+class DeployIdTest extends FlatSpec with Matchers {
+
+  private val runtimeManager: Resource[Task, RuntimeManager[Task]] =
+    mkRuntimeManager("deploy-id-runtime-manager-test")
+
+  private val sk = PrivateKey(
+    Base16.unsafeDecode("b18e1d0045995ec3d010c387ccfeb984d783af8fbb0f40fa7db126d889f6dadd")
+  )
+
+  private def deploy(
+      deployer: PrivateKey,
+      rho: String,
+      timestamp: Long = System.currentTimeMillis()
+  ): DeployData = ConstructDeploy.sourceDeploy(
+    source = rho,
+    timestamp = System.currentTimeMillis(),
+    accounting.MAX_VALUE,
+    sec = deployer
```

Comment:
please use sourceDeployNowF in a monadic context, and use the variant of the method that has the default parameter values hardcoded

Reply:
For others reading: this comment is invalid because that function hasn't landed in dev yet :D